test/object_storage/test_google.py

```diff
@@ -518,14 +523,127 @@ def test_delete_keys(preserve_trailing_slash: Union[bool, None]) -> None:
+        (100, 1),
+        (101, 2),
+        (200, 2),
+        (201, 3),
+        (1_000, 10),
+    ),
+)
+def test_delete_keys_bulk(total_keys: int, expected_bulk_request_count: int) -> None:
+    notifier = MagicMock()
+    test_keys = _generate_keys(total_keys)
+    with ExitStack() as stack:
+        stack.enter_context(patch("rohmu.object_storage.google.get_credentials"))
+        stack.enter_context(patch("rohmu.object_storage.google.GoogleTransfer._create_object_store_if_needed_unwrapped"))
+        mock_retry_on_reset = stack.enter_context(patch("rohmu.object_storage.google.GoogleTransfer._retry_on_reset"))
+        transfer = GoogleTransfer(
+            project_id="test-project-id",
+            bucket_name="test-bucket",
+            notifier=notifier,
+        )
+        mock_client = stack.enter_context(patch.object(transfer, "_object_client"))
+        mock_request = _mock_request([], resumable=None)
+        mock_client.return_value.__enter__.return_value.delete.return_value = mock_request
+
+        transfer.delete_keys(keys=test_keys)
+
+        assert mock_retry_on_reset.call_count == expected_bulk_request_count
+
+
+class BatchReqeustProcessor:
+    def __init__(self, callback: Callable[[str, HttpRequest | None, HttpError | None], None]) -> None:
+        self.callback = callback
+        self.operations: list[str] = []
+
+    def execute(self) -> None:
+        for key in self.operations:
+            if "500" in key:
+                self.callback(key, None, HttpError(resp=httplib2.Response({"status": "500"}), content=b"500"))
+                return
+            if "404" in key:
+                self.callback(key, None, HttpError(resp=httplib2.Response({"status": "404"}), content=b"404"))
+                return
+            self.callback(key, None, None)
+        return
+
+    def add(self, operation: HttpRequest, request_id: str) -> None:
+        self.operations.append(request_id)
+
+
+def test_delete_keys_callback() -> None:
+    notifier = MagicMock()
+    with ExitStack() as stack:
+        stack.enter_context(patch("rohmu.object_storage.google.get_credentials"))
+        stack.enter_context(patch("rohmu.object_storage.google.GoogleTransfer._create_object_store_if_needed_unwrapped"))
+        mock_delete_key = stack.enter_context(patch("rohmu.object_storage.google.GoogleTransfer.delete_key"))
+
+        transfer = GoogleTransfer(
+            project_id="test-project-id",
+            bucket_name="test-bucket",
+            notifier=notifier,
+        )
+
+        mock_batch_request = MagicMock()
+        mock_batch_request.execute.return_value = None
+
+        test_keys = ["key1", "key2", "key3_500"]
+
+        mock_gs = MagicMock()
+        mock_gs.new_batch_http_request = lambda callback: BatchReqeustProcessor(callback)
+        transfer.gs = mock_gs
+
+        transfer.delete_keys(keys=test_keys)
+
+        expected_calls = []
+        for key in ["key1", "key2"]:
+            expected_calls.extend([call(key)])
```

Comment:
Why can't you simply do the following?
```suggestion
        expected_calls = [call("key1"), call("key2")]
```

Reply:
Or even skip the variable definition and say `notifier.object_deleted.assert_has_calls([call("key1"), call("key2")])`